src/SoftRobots.Inverse/component/constraint/BarycentricCenterEffector.h

```diff
@@ -118,24 +109,31 @@ class BarycentricCenterEffector : public Effector<DataTypes>
     using Effector<DataTypes>::d_constraintIndex ;
     using Effector<DataTypes>::f_listening ;
     using Effector<DataTypes>::getTarget ;
-
+    using PositionEffector<DataTypes>::d_useDirections ;
+    using PositionEffector<DataTypes>::d_directions ;
+    using PositionEffector<DataTypes>::d_effectorGoal ;
+    using PositionEffector<DataTypes>::d_delta ;
+    using PositionEffector<DataTypes>::d_weight ;
+    using PositionEffector<DataTypes>::d_indices ;
+    using PositionEffector<DataTypes>::d_componentState ;
+
+    SOFA_ATTRIBUTE_DEPRECATED("v24.12", "v25.06", "Use d_useDirections instead.")
     sofa::Data<Vec<3,bool> > d_axis;
+
+    SOFA_ATTRIBUTE_DISABLED("v24.06", "v24.12", "Use d_effectorGoal instead.")
     sofa::Data<Coord >       d_effectorGoalPosition;
```

Comment:
@alxbilger would you mind having a look? I'm not sure about how I handle the deprecation :
1. `axis` is now `useDirections` from the user point of view. In `init()`, I check if the user used `axis`, print a message in that case, and fill `useDirections` with the user input.
2. `effectorGoal` is the same from the user point of view. It only changes for the developers. That's why I went directly for `SOFA_ATTRIBUTE_DISABLED`, but I'm not sure.

Reply:
`SOFA_ATTRIBUTE_DISABLED` does nothing. It has to be used with the type `DeprecatedAndRemoved`:

```cpp
SOFA_ATTRIBUTE_DISABLED("v24.06", "v24.12", "Use d_effectorGoal instead.")
DeprecatedAndRemoved d_effectorGoalPosition;
```